tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -133,11 +132,11 @@ def __init__(self, metadata: TestSuiteMetadata) -> None:
     def _initialize(self, node: Node, variables: Dict[str, Any], lisa_working_path: str, lisa_log_path: str, lisa_log: Logger):
         connection_info = node.connection_info
         node_context = get_node_context(node)
-        runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
 
         self.__context = self._Context(
             vm=VmIdentifier(
-                location=runbook.location,
+                cloud=self._get_required_parameter(variables, "c_cloud"),
```

Comment:
Now the AgentTestSuiteCombinator also generates the cloud, as "c_cloud"; for consistency, I am also picking up the location, as "c_location", from the values generated by the combinator, instead of from the runbook.